### PR TITLE
feature: generalize grasp_arms under the name base_offset

### PR DESCRIPTION
**Pull request by LarsJanssenTUe**

### CODE REVIEW

File: robot_smach_states/src/robot_smach_states/util/designators/ed_designators.py [hunk 1]
```diff
@@ -364,7 +364,13 @@ def is_poi_occupied(self, frame_stamped):
...
 
     def distance_to_poi_area(self, frame_stamped):
-        radius = math.hypot(self.robot.grasp_offset.x, self.robot.grasp_offset.y)
+
+        # ToDo: cook up something better: we need the arm that we're currently using but this would require a
+        # rather large API break
+        base_offset = self.robot.arms.values()[0].base_offset
+        radius = math.hypot(base_offset.x(), base_offset.y())
+
+        # radius = math.hypot(self.robot.grasp_offset.x, self.robot.grasp_offset.y)
```
Comment (jlunenburg) [inline robot_smach_states/src/robot_smach_states/util/designators/ed_designators.py]: Remove commented line

File: robot_skills/src/robot_skills/arms.py [hunk 2]
```diff
@@ -298,21 +298,21 @@ def __init__(self, robot_name, tf_listener, side):
...
+        offset = self.load_param('skills/arm/' + self.side + '/grasp_offset/')
         self.offset = kdl.Frame(kdl.Rotation.RPY(offset["roll"], offset["pitch"], offset["yaw"]),
                                 kdl.Vector(offset["x"], offset["y"], offset["z"]))
 
-        self.marker_to_grippoint_offset = self.load_param('skills/arm/offset/marker_to_grippoint')
+        self.marker_to_grippoint_offset = self.load_param('skills/arm/' + self.side + '/marker_to_grippoint')
+
+        # Grasp offsets
+        go = self.load_param('skills/arm/' + self.side + '/base_offset')
+        self._base_offset = kdl.Vector(go.get("x"), go.get("y"), go.get("z"))
```
Comment (LoyVanBeek) [inline robot_skills/src/robot_skills/arms.py]: Why the `.get`? If it's a dictionary, you can also do the `['x']` or `.get('x', default_val)`. If a `kdl.Vector` needs x,y,z and the dict provides those three, you could even do `kdl.Vector(**go)`.

File: robot_smach_states/src/robot_smach_states/util/designators/ed_designators.py [hunk 3]
```diff
@@ -364,7 +364,12 @@ def is_poi_occupied(self, frame_stamped):
         return not any(entities_at_poi)
 
     def distance_to_poi_area(self, frame_stamped):
-        radius = math.hypot(self.robot.grasp_offset.x, self.robot.grasp_offset.y)
+
+        # ToDo: cook up something better: we need the arm that we're currently using but this would require a
```
Comment (LoyVanBeek) [inline robot_smach_states/src/robot_smach_states/util/designators/ed_designators.py]: Can you make an issue for this an link to it here?

**Comment by LarsJanssenTUe** [comment]
Tested and in working state in the simulator!

File: robot_smach_states/src/robot_smach_states/navigation/navigate_to_grasp.py [hunk 4]
```diff
@@ -35,20 +35,8 @@ def generateConstraint(self):
...
-            elif arm == self.robot.arms['right']:
-                angle_offset = math.atan2(self.robot.grasp_offset.y, self.robot.grasp_offset.x)
-        elif self.robot.robot_name == "hero":
-            if arm == self.robot.arms['left']:
-                angle_offset = math.atan2(-self.robot.grasp_offset.y, self.robot.grasp_offset.x)
-            elif arm == self.robot.arms['right']:
-                angle_offset = math.atan2(-self.robot.grasp_offset.y, self.robot.grasp_offset.x)
-
-        radius = math.hypot(self.robot.grasp_offset.x, self.robot.grasp_offset.y)
+        angle_offset = -math.atan2(arm.base_offset.y(), arm.base_offset.x())
```
[the remark this answers is not in the document]
Comment (LarsJanssenTUe) [reply robot_smach_states/src/robot_smach_states/navigation/navigate_to_grasp.py]: Minus is incorrect.

**Pull request merged**